Glyssen/BookScript.cs

```diff
@@ -102,36 +103,50 @@ public System.Collections.Generic.IReadOnlyList<Block> GetScriptBlocks(bool join
 			if (!join || m_blockCount == 0)
 				return m_blocks;
 
-			var list = new List<Block>(m_blockCount);
-			list.Add(m_blocks[0]);
-			for (int i = 1; i < m_blockCount; i++)
+			var list = new List<Block>(m_blockCount) {m_blocks[0]};
+
+			for (var i = 1; i < m_blockCount; i++)
 			{
 				var block = m_blocks[i];
+				var prevBlock = list.Last();
+
 				if (!block.IsParagraphStart)
```

Comment:
Now that setting prevBlock is pulled outside, these two if statements could be combined into 1. Since the body of the following if is also the same, they could all be combined into one if it didn't make it too unreadable.